From 5fbcc73297e0fa8361f617b5f659c3f917e8a340 Mon Sep 17 00:00:00 2001
From: Dien Pham <dien.pham.ry@renesas.com>
Date: Thu, 12 Apr 2018 16:19:39 +0700
Subject: [PATCH 034/909] soc: renesas: rcar-sysc: Add SYSCEXTMASK info for
 r8a77990

commit bf1a7c8b3eca82f26eaf7122958f2242db0972cc from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Add specific SYSCEXTMASK register info so that
it is controlled during power domain on/off.

Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/r8a77990-sysc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/soc/renesas/r8a77990-sysc.c b/drivers/soc/renesas/r8a77990-sysc.c
index 15579ebc5ed2..ac2069e1964f 100644
--- a/drivers/soc/renesas/r8a77990-sysc.c
+++ b/drivers/soc/renesas/r8a77990-sysc.c
@@ -5,6 +5,7 @@
  * Copyright (C) 2018 Renesas Electronics Corp.
  */
 
+#include <linux/bitops.h>
 #include <linux/bug.h>
 #include <linux/kernel.h>
 #include <linux/sys_soc.h>
@@ -61,8 +62,13 @@ static int __init r8a77990_sysc_init(void)
 	return 0;
 }
 
+static struct rcar_sysc_extra_regs r8a77990_extra_regs = {
+	.sysc_extmask_offs = 0x2F8, .sysc_extmask_msks = BIT(0)
+};
+
 const struct rcar_sysc_info r8a77990_sysc_info __initconst = {
 	.init = r8a77990_sysc_init,
 	.areas = r8a77990_areas,
 	.num_areas = ARRAY_SIZE(r8a77990_areas),
+	.extra_regs = &r8a77990_extra_regs,
 };
-- 
2.17.1

